highfidelity/hifi
# Case 20700: Add missing pulse properties to certain Entity types

https://highfidelity.manuscript.com/f/cases/20700/Add-missing-color-alpha-pulse-properties-to-entities

Adds a new group of properties - "pulse" - to Shape, Image, Text, Web, Grid, and ParticleEffect entities.

These properties start off as deprecated.  The need for them is an unfortunate result of their existence in Overlay properties, but it's not a feature that was ever fully planned out or needed.  They could be removed at some point.

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/d1016d3a160d325d41bb840ed34cf535/raw/ae46cd408c3fdac11e4005c2d63a3429865de7b2/PulseTest.js).  You'll see:
![hifi-snap-by-samgondelman-on-2019-01-18_12-52-38](https://user-images.githubusercontent.com/7650116/51412409-5924d400-1b20-11e9-8691-1f6172e5672b.gif)
- Press "o".  The colors will pulse a little less:
![hifi-snap-by-samgondelman-on-2019-01-18_12-53-00](https://user-images.githubusercontent.com/7650116/51412422-63df6900-1b20-11e9-9b5f-d42ed92843d8.gif)
- Press "i".  The alpha will pulse instead of the color:
![hifi-snap-by-samgondelman-on-2019-01-18_12-53-11](https://user-images.githubusercontent.com/7650116/51412435-6d68d100-1b20-11e9-9bc2-b31d8c3212e3.gif)
- Press "o".  The alpha will pulse more:
![hifi-snap-by-samgondelman-on-2019-01-18_12-53-22](https://user-images.githubusercontent.com/7650116/51412444-748fdf00-1b20-11e9-9e7a-3a0407bf8774.gif)
